api/all/src/main/java/io/opentelemetry/api/common/AnyValue.java

```diff
@@ -105,6 +108,9 @@ static AnyValue<List<KeyAnyValue>> of(Map<String, AnyValue<?>> value) {
    * Return a string encoding of this {@link AnyValue}. This is intended to be a fallback serialized
    * representation in case there is no suitable encoding that can utilize {@link #getType()} /
    * {@link #getValue()} to serialize specific types.
+   *
+   * <p>WARNING: No guarantees are made about the encoding of this string response. It MAY change in
+   * a future minor release. If you need a reliable string encoding, write your own serializer.
    */
   // TODO(jack-berg): Should this be a JSON encoding?
```

Comment:
The closest we have is https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/attribute-type-mapping.md, which explains how to map data structures to AnyValue, but nothing about a string representation of AnyValue. If the spec did have something, I imagine it might either draw inspiration from or be defined as the [standard protobuf JSON mapping](https://protobuf.dev/programming-guides/proto3/#json).

I agree that a stable spec would be ideal. But I also think we can proceed without it. Ideally, exporters quickly update to be able to properly serialize AnyValue bodies. The goal of `asString()` is just to be a fallback allowing the information to be transmitted, but in a format which implies that you shouldn't depend on it. Seeing data in this format is a signal that something is suboptimal in the export pipeline, while still allowing things to work.

If you look at a test of [AnyValue#asString](https://github.com/open-telemetry/opentelemetry-java/blob/26b3d41d721e1e6cd054659760f902252247393c/api/incubator/src/test/java/io/opentelemetry/api/incubator/logs/AnyValueTest.java#L169-L212), you see a format which is JSON-like, but definitely not actually JSON. The format is probably best described as an idiomatic implementation of Java toString(). It could easily be adapted to be actual valid JSON, but that might reduce the incentive for exporters to properly serialize.